### PR TITLE
edit default checkpoint path

### PR DESCRIPTION
**Pull request by rockb1017**
To make deployment in Autopilot work out of the box

### CODE REVIEW

**Comment by dmitryax** [comment]
We also need to add a copy command in the initContainer to avoid re-reading all the logs on upgrade. There are few Observability customers using the native logs collection

**Comment by rockb1017** [comment]
> We also need to add a copy command in the initContainer to avoid re-reading all the logs on upgrade. There are few Observability customers using the native logs collection

hmm so mount both directories and copy files if it doesn't already exist in the new path?

**Comment by rockb1017** [comment]
I am using `cp -n` for no overwrite. and making sure it always exit 0 with ` || true`
```
-n, --no-clobber
              do not overwrite an existing file (overrides a previous -i option)
```

File: helm-charts/splunk-otel-collector/templates/daemonset.yaml [hunk 1]
```diff
@@ -94,6 +94,19 @@ spec:
             - name: fluentd-config-cri
               mountPath: /fluentd/etc/cri
         {{- else }}
+        - name: copy-old-checkpoint
+          image: {{ template "splunk-otel-collector.image.fluentd" . }}
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/daemonset.yaml:98]: Let's use splunk-otel-collector image here so we don't pull an extra image if native logs collection enabled

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,7 +15,11 @@ guidelines](https://github.com/signalfx/splunk-otel-collector-chart/blob/main/UP
 
 - Extract `container.image.tag` attribute from `container.image.name` (#285)
 - Upgrade splunk-otel-collector image to 0.38.1 (#284)
+<<<<<<< HEAD
 - Change secret names according to the GDI specification (#295)
+=======
+- Changed the default checkpoint path to `/var/addon/splunk/otel_pos` (#292)
+>>>>>>> f8979ec8b53d (add init container for copying old checkpoint)
```
Comment (dmitryax) [inline CHANGELOG.md:22]: Please resolve

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
@rockb1017 Follow-up: we have the `checkpointPath` configuration option but all the initContainers use hardcoded `/var/addon/splunk/otel_pos` value instead. 

I believe changing `checkpointPath` breaks the deployment. We need to decide how to solve it:
1. Make sure `checkpointPath` is used in all initContainers
2. Remove `checkpointPath` option and use hardcoded values instead.

I vote for 2. We can always bring that configuration back if there will be a need. Please let me know what do you think

**Pull request merged**